src/libs/ReportUtils.ts

```diff
@@ -2414,48 +2435,51 @@ function updateOptimisticParentReportAction(parentReportAction, lastVisibleActio
 
 /**
  * Get optimistic data of parent report action
- * @param {String} reportID The reportID of the report that is updated
- * @param {String} lastVisibleActionCreated Last visible action created of the child report
- * @param {String} type The type of action in the child report
- * @param {String} parentReportID Custom reportID to be updated
- * @param {String} parentReportActionID Custom reportActionID to be updated
- * @returns {Object}
- */
-function getOptimisticDataForParentReportAction(reportID, lastVisibleActionCreated, type, parentReportID = '', parentReportActionID = '') {
+ * @param reportID The reportID of the report that is updated
+ * @param lastVisibleActionCreated Last visible action created of the child report
+ * @param type The type of action in the child report
+ * @param parentReportID Custom reportID to be updated
+ * @param parentReportActionID Custom reportActionID to be updated
+ */
+function getOptimisticDataForParentReportAction(reportID: string, lastVisibleActionCreated: string, type: string, parentReportID = '', parentReportActionID = ''): OnyxUpdate | EmptyObject {
     const report = getReport(reportID);
+    if (!report || !isNotEmptyObject(report)) {
```

Comment:
Any reason why we're using `!isNotEmptyObject` instead of `isEmptyObject`?

Reply:
Better type narrowing

Reply:
I think `isEmptyObject` has the same type narrowing capabilities. @youssef-lr it may be a mistake, `isEmptyObject` seems better